packages/search/src/models.ts

```diff
@@ -0,0 +1,25 @@
+export type PrimitiveType = string | number | boolean;
+
+/**
+ * Generic Search fields type.
+ * @public
+ */
+export type GenericFields = {
```

Comment:
I think name is a bit confusing for API users, like what generic, what fields. Lets re-exports this type as something domain specific like `SearchDocument`, `DocumentModel` etc.